dotCMS/src/main/java/com/dotcms/variant/business/VariantCacheImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.dotcms.variant.business;
+
+import com.dotcms.util.DotPreconditions;
+import com.dotcms.variant.model.Variant;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.business.DotCacheException;
+
+public class VariantCacheImpl implements VariantCache{
+
+    static final String VARIANT_BY_ID = "VARIANT_BY_ID_";
```

Comment:
consider to handle the 404

Reply:
done https://github.com/dotCMS/core/pull/22912/commits/8b4f3aad6a9123079e07d3e2e9a47bfc14d152f1